service/matching/matching_engine.go

```diff
@@ -1372,34 +1378,99 @@ func (e *matchingEngineImpl) unloadTaskQueuePartition(unloadPM taskQueuePartitio
 	delete(e.partitions, key)
 	e.partitionsLock.Unlock()
 	foundTQM.Stop()
-	e.updateTaskQueueGauge(foundTQM, false, -1)
 }
 
-func (e *matchingEngineImpl) updateTaskQueueGauge(pm taskQueuePartitionManager, versioned bool, delta int) {
-	countKey := taskQueueCounterKey{
+// Responsible for emitting and updating loaded_physical_task_queue_count metric
+func (e *matchingEngineImpl) updatePhysicalTaskQueueGauge(pm *physicalTaskQueueManagerImpl, delta int) {
+
+	// calculating versioned to be one of: “unversioned” or "buildId” or “versionSet”
+	versioned := "unversioned"
+	if buildID := pm.queue.BuildId(); buildID != "" {
+		versioned = "buildId"
+	} else if versionSet := pm.queue.VersionSet(); versionSet != "" {
+		versioned = "versionSet"
+	}
+
+	PhysicalTaskQueueParameters := taskQueueCounterKey{
+		namespaceID:   pm.partitionMgr.Partition().NamespaceId(),
+		taskType:      pm.partitionMgr.Partition().TaskType(),
+		partitionType: pm.partitionMgr.Partition().Kind(),
+		versioned:     versioned,
+	}
+
+	e.gaugeMetricCounter.gaugeMetricCounterLock.Lock()
+	e.gaugeMetricCounter.loadedPhysicalTaskQueueCount[PhysicalTaskQueueParameters] += delta
+	loadedPhysicalTaskQueueCounter := e.gaugeMetricCounter.loadedPhysicalTaskQueueCount[PhysicalTaskQueueParameters]
+	e.gaugeMetricCounter.gaugeMetricCounterLock.Unlock()
+
+	pmImpl := pm.partitionMgr
+
+	e.metricsHandler.Gauge(metrics.LoadedPhysicalTaskQueueCounter.Name()).Record(
+		float64(loadedPhysicalTaskQueueCounter),
+		metrics.NamespaceTag(pmImpl.ns.Name().String()),
+		metrics.TaskTypeTag(PhysicalTaskQueueParameters.taskType.String()),
+		metrics.PartitionTypeTag(PhysicalTaskQueueParameters.partitionType.String()),
+	)
+}
+
+// Responsible for emitting and updating loaded_task_queue_family_count, loaded_task_queue_count and
+// loaded_task_queue_partition_count metrics
+func (e *matchingEngineImpl) updateTaskQueuePartitionGauge(pm taskQueuePartitionManager, delta int) {
+
+	// each metric shall be accessed based on the mentioned parameters
+	TaskQueueFamilyParameters := taskQueueCounterKey{
```

Comment:
same here and lots of other places
```suggestion
	taskQueueFamilyParameters := taskQueueCounterKey{
```